src/fortran/client.F90

```diff
@@ -1444,4 +1464,259 @@ function use_tensor_ensemble_prefix(self, use_prefix) result(code)
+  integer,              intent(in)  :: list_length !< The desired length of the list
+  integer,              intent(in)  :: poll_frequency_ms  !< Frequency at which to poll the database (ms)
+  integer,              intent(in)  :: num_tries          !< Number of times to poll the database before failing
+  logical(kind=c_bool), intent(out) :: poll_result !< True if the list is the requested length, False if not after num_tries.
+  integer(kind=enum_kind)          :: code
+  
+  ! Local variables
+  character(kind=c_char, len=len_trim(list_name)) :: list_name_c
+  integer(kind=c_size_t) :: list_name_length
+  integer(kind=c_int) :: c_poll_frequency, c_num_tries, c_list_length
+
+  list_name_c = trim(list_name)
+  list_name_length = len_trim(list_name)
+  c_num_tries = num_tries
+  c_poll_frequency = poll_frequency_ms
+  c_list_length = list_length
+
+  code = poll_list_length_gte_c(self%client_ptr, list_name_c, list_name_length, &
+                            c_list_length, c_poll_frequency, c_num_tries, poll_result)
+end function poll_list_length_gte
+
+!> Get the length of the aggregation list
+function poll_list_length_lte(self, list_name, list_length, poll_frequency_ms, num_tries, poll_result) result(code)
+  class(client_type),   intent(in) :: self       !< An initialized SmartRedis client
+  character(len=*),     intent(in) :: list_name    !< Name of the dataset to get
+  integer,              intent(in)  :: list_length !< The desired length of the list
+  integer,              intent(in)  :: poll_frequency_ms  !< Frequency at which to poll the database (ms)
+  integer,              intent(in)  :: num_tries          !< Number of times to poll the database before failing
+  logical(kind=c_bool), intent(out) :: poll_result !< True if the list is the requested length, False if not after num_tries.
+  integer(kind=enum_kind)          :: code
+  
+  ! Local variables
+  character(kind=c_char, len=len_trim(list_name)) :: list_name_c
+  integer(kind=c_size_t) :: list_name_length
+  integer(kind=c_int) :: c_poll_frequency, c_num_tries, c_list_length
+
+  list_name_c = trim(list_name)
+  list_name_length = len_trim(list_name)
+  c_num_tries = num_tries
+  c_poll_frequency = poll_frequency_ms
+  c_list_length = list_length
+
+  code = poll_list_length_lte_c(self%client_ptr, list_name_c, list_name_length, &
+                            c_list_length, c_poll_frequency, c_num_tries, poll_result)
+end function poll_list_length_lte
+
+!> Get datasets from an aggregation list. Note that this will deallocate an existing list
+function get_datasets_from_list(self, list_name, datasets, num_datasets) result(code)
+  class(client_type),   intent(in) :: self       !< An initialized SmartRedis client
+  character(len=*),     intent(in) :: list_name    !< Name of the dataset to get
+  type(dataset_type), dimension(:), allocatable, intent(  out) :: datasets !< The array of datasets included
```

Comment:
( out) -- while I like this convention of putting the space to show the direction, you haven't been doing it consistently